packages/swing-store/src/archiver.js

```diff
@@ -0,0 +1,44 @@
+import { finished as streamFinishedCallback, Readable } from 'node:stream';
+import { promisify } from 'node:util';
+import { createGzip } from 'node:zlib';
+import { withDeferredCleanup } from '@agoric/internal/src/node/utils.js';
+
+const streamFinished = promisify(streamFinishedCallback);
+
+/*
+ * @param {string} dirPath
+ * @param {object} powers
+ * @param {Pick<import('fs'), 'createWriteStream' | 'mkdirSync' | 'renameSync'>} powers.fs
+ * @param {Pick<import('path'), 'join'>} powers.path
+ * @param {Pick<import('tmp'), 'fileSync'>} powers.tmp
+ */
+export const makeArchiveTranscript = (dirPath, powers) => {
+  const { fs, path, tmp } = powers;
+  fs.mkdirSync(dirPath, { recursive: true });
+  const archiveTranscript = (spanName, entries) => {
+    const destPath = path.join(dirPath, `${spanName}.gz`);
+    return withDeferredCleanup(async addCleanup => {
+      const {
+        name: tmpName,
+        fd,
+        removeCallback,
+      } = tmp.fileSync({
```

Comment:
Why use the sync version if we're in an async function ?

Reply:
No strong reason; this step doesn't need to be async and honestly wouldn't benefit from it anyway.

Reply:
I'm just wondering because I believe it means the removeCallback is sync, which may be unexpected (according to types) for the cleanup. I do remember that `tmp` had some surprising changes in behavior between sync and async, but I'm not sure what those where now.